server/routes.js

```diff
@@ -1,16 +1,39 @@
 const router = require('express').Router();
+const cache = require('memory-cache');
 const controllers = require('./controllers');
 
-router.get('/averageReviews/:itemId', controllers.averageReviews.get);
+const memCache = new cache.Cache();
 
-router.get('/reviews/:itemId', controllers.reviews.get);
+const cacheMiddleware = (duration) => {
+  return (req, res, next) => {
+    // eslint-disable-next-line prefer-template
+    const key = '__express__' + req.originalUrl || req.url;
+    const cacheContent = memCache.get(key);
+
+    if (cacheContent) {
+      res.send(JSON.parse(cacheContent));
```

Comment:
JSON parse should be in a try/catch because it throws an error, which will stop execution. Catch your errors :)

Reply:
corrected.  Will be in PR for production branch.

Reply:
btw you can also do middleware for catching errors. Instead of doing it per route like you did the caching middleware, you can have with `app.use(yourErrorMiddleWare)`  before your routes, and it'll also catch errors from your caching middleware. For errors from the db or other async calls you ccan use use the `next(err)` instead of specifically writing your own middleware